src/os_posix/os_yield.c

```diff
@@ -24,3 +24,29 @@ __wt_yield(void) WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
 
     sched_yield();
 }
+
+/*
+ * If HAVE_DIAGNOSTIC is 0 clang will raise an error when the noreturn attribute is missing, and if
+ * HAVE_DIAGNOSTIC is 1 clang will raise an error when the noreturn attribute is present. This
+ * function should never be called when HAVE_DIAGNOSTIC is 0 so we can ignore the missing-noreturn
+ * warning here.
+ */
+#pragma GCC diagnostic push
```

Comment:
Suppressing `missing-noreturn` here isn't my favourite solution but it's the cleanest fix I've found so far.
Any ideas to improve on this are very welcome.

Reply:
If we have two separate function definitions would it stop complaining:

#ifndef HAVE_DIAGNOSTIC
void
__wt_yield_no_barrier(void) WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
{
    __wt_abort(NULL);
}
#else
void
__wt_yield_no_barrier(void) WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
{
    sched_yield();
}
#endif

Reply:
We hit some issues with `s_all` in that case. 
`python prototypes.py` adds both function signatures in `extern_posix.h` that only differ by the `noreturn` attribute, which causes a `-Wredundant-decls` compilation error

Reply:
That is annoying. I will LGTM, but maybe we should fix s_all ?